docs/cookbook/production-deployment.md

```diff
@@ -0,0 +1,248 @@
+# Production Deployment
+
+## Server Setup
+
+CentOS is very common OS for production servers so we use it on our production server.
+
+### Docker
+
+First we install Docker.
+```
+yum -y install docker-ce
+```
+We set docker to run as a service.
+```
+systemctl enable docker-ce
+systemctl start docker
+```
+
+### Firewall
+
+It is very important to have our containers inaccessible from outside. 
+For that purpose we need to update `firewalld` configuration with these commands, 
+because Docker overrides `firewalld` and publishes ports on the server by default.
+```
+firewall-cmd --permanent --direct --add-chain ipv4 filter DOCKER
+firewall-cmd --permanent --direct --add-rule ipv4 filter DOCKER 0 ! -s 127.0.0.1 -j RETURN
+```
+
+### Nginx
+
+Let's presume that we want to have our site running on `HTTPS` protocol and everything that concerns domain and its `certificates` is already setup.
+Only thing that is missing is to connect the domain to application that runs in docker containers via port 8000 on 127.0.0.1 ip address.
+First we need to allow Nginx to connect to sockets by executing a command in shell console.
+ ```
+setsebool httpd_can_network_connect on -P
+```
+Then we add location block into `/etc/nginx/conf.d/<YOUR_DOMAIN_HERE>.conf` into server block.
+ ```
+location / {
+    proxy_set_header Host $host;
+    proxy_set_header X-Real-IP $remote_addr;
+    proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
+    proxy_set_header X-Forwarded-Proto $https;
+    proxy_pass http://127.0.0.1:8000;
+}
+```
+
+### Database
+
+We need PostgresSQL in version 10.5 installed. To get this done we need to add repository to our server and then install PostgresSQL server with PostgresSQL client.
+```
+rpm -Uvh https://yum.postgresql.org/10/redhat/rhel-7-x86_64/pgdg-centos10-10-2.noarch.rpm
+yum install postgresql10-server postgresql10 postgresql10-contrib
+```
+Next we initialise PostgresSQL database.
+ ```
+/usr/pgsql-10/bin/postgresql-10-setup initdb
+```
+We need to allow access to database from docker network that will operate on 192.168.0.0 subnet.
+```
+echo `host all all 192.168.0.1/16 md5` > /var/lib/pgsql/10/data/pg_hba.conf
+```
+We edit configuration file `/var/lib/pgsql/10/data/postgresql.conf` of postgresql to match application needs based on our [postgres.conf](../../docker/postgres/postgres.conf).
+Now we register and launch PostgresSQL server as a service. 
+```
+systemctl start postgresql-10
+systemctl enable postgresql-10
+```
+Next we create new database user with login `root` using default administrator user that will be used by application.
+```
+su postgres -c "createuser --createdb --superuser --pwprompt root"
```

Comment:
I cannot move throw this part, I ended with error
createuser: nelze navázat spojení s databází postgres: FATAL:  no pg_hba.conf entry for host "[local]", user "postgres", database "postgres", SSL off

Reply:
problem was with the rewriting of /var/lib/pgsql/10/data/pg_hba.conf
now it is >> instead of > so it will add one more line and will not remove the original data.